.yamato/ruamel/editor_pinning/store_green_revisions.py

```diff
@@ -0,0 +1,174 @@
+from util.subprocess_helpers import git_cmd, run_cmd
+
+yaml = ruamel.yaml.YAML()
+
+
+def load_yml(filepath):
+    with open(filepath) as f:
+        return yaml.load(f)
+
+def ordereddict_to_dict(d):
+    return {k: ordereddict_to_dict(v) for k, v in d.items()} if isinstance(d, OrderedDict) else d
+
+
+def checkout_and_pull_branch(branch, working_dir, development_mode=False):
+    if not development_mode:
+        git_cmd(f'checkout {branch}', working_dir)
+        git_cmd('pull', working_dir)
+
+def commit_and_push(commit_msg, working_dir, development_mode=False):
+    if not development_mode:
+        git_cmd(['commit', '-m', commit_msg], working_dir)
+        git_cmd('pull', working_dir)
+        git_cmd('push', working_dir)
+
+def get_last_nightly_id(api_key):
+    try:
+        current_date = str(datetime.date.today())
+        url = f"http://yamato-api.cds.internal.unity3d.com/jobs?filter=project eq 902 and branch eq '10.x.x/release' and filename eq '.yamato%252F_abv.yml%2523all_project_ci_nightly_2020.2' and submitted gt '{current_date}'"
+        headers={"Authorization":f"ApiKey {api_key}"}
+        response = requests.get(url=url, headers=headers)
+        if response.status_code != 200:
+            raise Exception()
+
+        jobs = response.json()["items"]
+        if len(jobs) == 0:
+            print(f"!! WARNING: No jobs found on {current_date}")
+            return None
+
+        for job in jobs:
+            if job["links"]["triggeredBy"] == "/users/0":
+                return job["id"]
+        print(f"!! WARNING: No jobs submitted by CI found on {current_date}")
+        return None
+
+    except:
+        print(f"!! ERROR: Failed to call Yamato API. Got {response.json()}")
+        return None
+
+
+def get_yamato_dependency_tree(job_id, api_key):
+    """Calls Yamato API (GET/jobid/tree)for given job id. Returns JSON dependency tree if success, and None if fails."""
+    try:
+        url = f'http://yamato-api.cds.internal.unity3d.com/jobs/{job_id}/tree'
+        headers={"Authorization":f"ApiKey {api_key}"}
+        response = requests.get(url=url, headers=headers)
+        dependency_tree = response.json()
+        if response.status_code != 200:
+            raise Exception()
+        return dependency_tree
+    except:
+        print(f"!! ERROR: Failed to call Yamato API. Got {response.json()}")
+        return None
+
+def update_green_project_revisions(editor_versions_file, project_versions_file, track, green_revision_jobs, job_id, api_key, working_dir):
+    """Updates green project revisions file for given track. If any updates present, adds to git and returns True. If not, returns False."""
+
+    # get the revisions used for the job, the last green project revisions, and Yamato dependency tree  
+    updated_at = str(datetime.datetime.utcnow())
+    revisions_key = f"{track}_latest_internal" if track=="trunk" else f"{track}_staging"
```

Comment:
is this reference to trunk meant to be here?

Reply:
yep, it just makes trunk use latest_internal and other tracks use staging